### ISSUE TITLE
Wrong LocalTime while converting from one timezone to another

### ISSUE THREAD

**Issue by zezekalo** [dlew/joda-time-android#191]
I used sample application and have just changed **sampleLocalDate** method:
```
`private void sampleLocalDate() {
        List<String> text = new ArrayList<String>();
        LocalDate now = LocalDate.now();
        text.add("Now: " + now);
        long startDayMillis = now.toDateTimeAtStartOfDay().getMillis();
        text.add("LocalDate in millis: " + startDayMillis);

        long startDayMillisInUTC = now.toDateTimeAtStartOfDay().withZone(DateTimeZone.UTC).getMillis();
        text.add("startDayMillisInUTC: " + startDayMillisInUTC);

        text.add("LocalDate after converting: " + new LocalDate(startDayMillis));
        text.add("Now + 2 days: " + now.plusDays(2));
        text.add("Now + 3 months: " + now.plusMonths(3));

        text.add("----------------------------------------------");

        LocalTime time = new LocalTime(DateTimeZone.UTC);
        text.add("LocalTime: " + time);
        long timeNowMillis = time.getMillisOfDay();
        text.add("LocalTime in millis - timeNowMillis: " + timeNowMillis);
        long millis = time.getMillisOfDay();
        LocalTime localTime = new LocalTime(millis, DateTimeZone.UTC);
        text.add("LocalTime, getMillisOfDay: " + millis + "; local time (UTC) - " + localTime);
        TimeZone timeZone = TimeZone.getDefault();
        text.add("Timezone: rawOffset: " + timeZone.getRawOffset()/(60*60*1000) + "; timezone - " + timeZone.toString());
        text.add("LocalTime, local time (device timezone) - "
                + new LocalTime(millis, DateTimeZone.getDefault())
//                + new DateTime(millis, DateTimeZone.forTimeZone(timeZone))
        );
        text.add("difference between timeNowMillis and startDayMillis: " + new LocalTime(timeNowMillis - startDayMillis, DateTimeZone.UTC));

        addSample("LocalDate", text);`
```

My idea is - I have LocalTime in UTC and save it as long in database. So, I can restore long timestamp and convert it in LocalTime in UTC. This works well. But I 'd like to convert LocalTime(UTC) to LocalTime(TimeZoneOfMyDevice).
I tried converting but I had an issue - converted LocalTime differ from current time

![device-2017-11-07-184846](https://user-images.githubusercontent.com/2778281/32507026-f53cb2e8-c3ee-11e7-9c16-e4a59bc23777.png)

What could be wrong in my case?
Device: Xiaomi Redmi Note 4X,

**Comment by dlew** [comment]
I'm not exactly sure what you're trying to do. If you're using `LocalTime` that means that you don't care about timezone at all, so to throw timezones into the mix confuses me.

**Comment by zezekalo** [comment]
Hmmm, but `LocalTime`-class  has the constructor which includes `DateTimeZone` parameter. If it means that `LocalTime` don't care about timezone then why it has `DateTimeZone` argument?

P.S. After a little investigation I tend to think that my case is in issue of MIUI 8 of Xiaomi. And it happens when I switch on 'Automatic time zone' in settings.

**Comment by dlew** [comment]
Sure, you can construct a `LocalTime` from a `DateTimeZone`, but afterwards you are essentially *losing* that data. That's useful when you, say, have a time picker that includes timezone: the user picks the time they want (11 AM in zone X), you convert that to `LocalTime` (11 AM anywhere), and the timezone doesn't matter anymore.

If you want to convert `LocalTime` back to `DateTime`, maybe you should use [`LocalTime.toDateTimeToday()`](http://joda-time.sourceforge.net/api-2.0/org/joda/time/LocalTime.html#toDateTimeToday())?

**Comment by zezekalo** [comment]
Yeah, thanks for your help, I think that this is what I want. But when I changed code to your advice then the time difference on my device is still present((
```
text.add("LocalTime, local time (device timezone) - "
                  + new LocalTime(millis, DateTimeZone.getDefault()).toDateTimeToday());
```
![device-2017-11-08-163159](https://user-images.githubusercontent.com/2778281/32554534-d3db9ec0-c4a2-11e7-80bb-8ec0053aaafe.png)

**Comment by dlew** [comment]
What timezone is your phone in, and how old is your phone? I'm beginning to suspect the problem may be that the zone data on your phone is out-of-date compared to joda-time. That could result in the output being wrong between the system clock and your app's clock.

**Comment by zezekalo** [comment]
At the screenshot above I switched on 'Automatic time zone' and I gave 'timezone' from my internet provider.
Yes, my real timezone is "Europe/Kyiv"

But when I switched off 'Automatic time zone' and selected manually 'Europe/Kyev' I found  such result:
![device-2017-11-08-163825](https://user-images.githubusercontent.com/2778281/32555069-6505750a-c4a4-11e7-81fa-0fc72f2e6801.png)

Bucarest? )) Chinese is such Chinese))

**Comment by zezekalo** [comment]
Hmmm, why do other time-applications work well in such case?))

**Comment by dlew** [comment]
I really don't know. I could look more into this if you provide a brief snippet that demonstrates the exact problem (and not just underlined areas of the sample app). Otherwise I don't really have time to chase this down; I'm guessing it's probably not the lib's fault, but some misunderstanding of the timezones involved.
